src/components/DistanceRequest.js

```diff
@@ -90,16 +89,17 @@ function DistanceRequest({iou, iouType, report, transaction, mapboxAccessToken})
 
     const lastWaypointIndex = numberOfWaypoints - 1;
     const isLoadingRoute = lodashGet(transaction, 'comment.isLoading', false);
-    const hasRouteError = lodashHas(transaction, 'errorFields.route');
+    const hasRouteError = !lodashIsNil(lodashGet(transaction, 'errorFields.route'));
     const haveWaypointsChanged = !_.isEqual(previousWaypoints, waypoints);
-    const doesRouteExist = lodashHas(transaction, 'routes.route0.geometry.coordinates');
+    const doesRouteExist = TransactionUtils.doesRouteExist(transaction)
     const validatedWaypoints = TransactionUtils.getValidWaypoints(waypoints);
-    const shouldFetchRoute = (!doesRouteExist || haveWaypointsChanged) && !isLoadingRoute && _.size(validatedWaypoints) > 1;
+    const isRouteAbsentWithNoErrors = !hasRouteError && !doesRouteExist;
+    const shouldFetchRoute = (haveWaypointsChanged || isRouteAbsentWithNoErrors) && !isLoadingRoute && _.size(validatedWaypoints) > 1;
     const waypointMarkers = useMemo(
         () =>
             _.filter(
                 _.map(waypoints, (waypoint, key) => {
-                    if (!waypoint || !lodashHas(waypoint, 'lat') || !lodashHas(waypoint, 'lng') || lodashIsNull(waypoint.lat) || lodashIsNull(waypoint.lng)) {
+                    if (!waypoint || lodashIsNil(lodashGet(waypoint, 'lng')) || lodashIsNil(lodashGet(waypoint, 'lat'))) {
```

Comment:
This specific change is only for cosmetic purposes to reflect my changes in here: https://github.com/Expensify/App/blob/9ce8d13d918cf6a0de89ae673c21dbc435a605b1/src/components/ConfirmedRoute.js#L47

And **that** fix was related not only to non-existing address. Without my change, going back online on the confirmation screen after entering any address points would crash the mobile app.

https://github.com/Expensify/App/assets/12595293/3cc4e401-5b8a-4daa-8976-982c95e5a135

Reply:
In this particular case we still did return early in the null cases – 4th and 5th checks: `... || lodashIsNull(waypoint.lat) || lodashIsNull(waypoint.lng)`.
However, maybe the fact that I've changed isNull to isNil now handles this case, because it covers `undefined` as well, not only `null`.